### PR TITLE
replaced +load with +initialized, optimized thread comparison

### PR DESCRIPTION
**Pull request by andrewgubanov**
Category has nothing specific to be done right after the class is added to runtime, but it does right before the first message to the object of this class is sent. So it is more appropriate to use +initialize in this case.
Also, there is no reason to call +[super load] as runtime has already sent this method to your subclass right before it called this method on your class.

I have also remove redundant code to compare current thread and main thread. In the previous code you have already made comparison if it was a main thread, and then still call -[NSThread currentThread] to get a pointer to current thread to yet compare it with the main thread. So i simplified this comparison.

Hope that helps.

### CODE REVIEW

**Comment by NSElvis** [comment]
>Category has nothing specific to be done right after the class is added to runtime, but it does right before the first message to the object of this class is sent. So it is more appropriate to use +initialize in this case.
Also, there is no reason to call +[super load] as runtime has already sent this method to your subclass right before it called this method on your class.

Interesting! @zenangst any comments on this?

>I have also remove redundant code to compare current thread and main thread. In the previous code you have already made comparison if it was a main thread, and then still call -[NSThread currentThread] to get a pointer to current thread to yet compare it with the main thread. So i simplified this comparison.

This might not apply anymore, I roll-back the decision of not allowing saves on the main thread

https://github.com/hyperoslo/NSManagedObjectContext-HYPSafeSave/pull/5

**Comment by zenangst** [comment]
@andrewgubanov from what I gather, it's best practice to always swizzle in `+load` and not `+initialize`. However, calling [super load] is just a brain-fart on my part (it's fun to rhyme) and that will be removed.

For more information, have a look at: http://nshipster.com/method-swizzling/

tl;dr

> Because method swizzling affects global state, it is important to minimize the possibility of race conditions. +load is guaranteed to be loaded during class initialization, which provides a modicum of consistency for changing system-wide behavior. By contrast, +initialize provides no such guarantee of when it will be executed—in fact, it may never be called, if that class is never messaged directly by the app.

**Comment by andrewgubanov** [comment]
So what is the point of methods swizzling if there is no call to that class at all?

**Comment by andrewgubanov** [comment]
Please let me know if you are still not convinced, so I will close the pull request :)

**Comment by 3lvis** [comment]
@andrewgubanov: you have a valid point, `swizzling` **AND** category give the wrong impression and contradict each other. We need to decide for one of them, so far `swizzling` sounds like a winner.

**Comment by andrewgubanov** [comment]
@zenangst this is pretty interesting article about more proper way of swizzling :) http://blog.newrelic.com/2014/04/16/right-way-to-swizzle/

**Comment by zenangst** [comment]
@andrewgubanov thanks, I'll have a look :)
